src/Product/Invoice/InvoiceV4Document.php

```diff
@@ -0,0 +1,214 @@
+     */
+    public Taxes $taxes;
+    /**
+     * @var AmountField
+     */
+    public AmountField $totalAmount;
+    /**
+     * @var AmountField
+     */
+    public AmountField $totalNet;
+
+    /**
+     * @param array        $rawPrediction Raw prediction from HTTP response.
+     * @param integer|null $pageId        Page number for multi pages PDF.
+     */
+    public function __construct(array $rawPrediction, ?int $pageId = null)
+    {
+        $this->customerAddress = new StringField($rawPrediction["customer_address"], $pageId);
+        $this->customerCompanyRegistrations = array_map(
+            fn ($prediction) => new CompanyRegistrationField($prediction, $pageId),
+            $rawPrediction["customer_company_registrations"]
+        );
+        $this->customerName = new StringField($rawPrediction["customer_name"], $pageId);
+        $this->date = new DateField($rawPrediction["date"], $pageId);
+        $this->documentType = new ClassificationField($rawPrediction["document_type"], $pageId);
+        $this->dueDate = new DateField($rawPrediction["due_date"], $pageId);
+        $this->invoiceNumber = new StringField($rawPrediction["invoice_number"], $pageId);
+        $this->lineItems = array_map(
+            fn ($prediction) => new InvoiceV4LineItems($prediction, $pageId),
+            $rawPrediction["line_items"]
+        );
+        $this->locale = new LocaleField($rawPrediction["locale"], $pageId);
+        $this->supplierCompanyRegistrations = array_map(
+            fn ($prediction) => new CompanyRegistrationField($prediction, $pageId),
+            $rawPrediction["supplier_company_registrations"]
+        );
+        $this->supplierName = new StringField($rawPrediction["supplier_name"], $pageId);
+        $this->supplierPaymentDetails = array_map(
+            fn ($prediction) => new PaymentDetailsField($prediction, $pageId),
+            $rawPrediction["supplier_payment_details"]
+        );
+        $this->taxes = new Taxes($rawPrediction['taxes'], $pageId);
+        $this->totalAmount = new AmountField($rawPrediction['total_amount'], $pageId);
+        $this->totalNet = new AmountField($rawPrediction['total_net'], $pageId);
+    }
+
+    /**
+     * Creates a line of rST table-compliant string separators.
+     *
+     * @param string $char Character to use as a separator.
+     * @return string
+     */
+    private static function lineItemsSeparator(string $char): string
+    {
+        $outStr = "  ";
+        $outStr .= "+" . str_repeat($char, 38);
+        $outStr .= "+" . str_repeat($char, 14);
+        $outStr .= "+" . str_repeat($char, 10);
+        $outStr .= "+" . str_repeat($char, 12);
+        $outStr .= "+" . str_repeat($char, 14);
+        $outStr .= "+" . str_repeat($char, 14);
+        $outStr .= "+" . str_repeat($char, 12);
+        return $outStr . "+";
+    }
+
+    /**
+     * String representation for line items.
+     *
+     * @return string
+     */
+    private function lineItemsToStr(): string
```

Comment:
suggest putting this in a new InvoiceV4LineItems class similar to the Taxes class.